modules/end-user-guide/pages/adding-a-vscode-extension.adoc

```diff
@@ -0,0 +1,163 @@
+:navtitle: Adding a Visual Studio Code extension to a workspace
+:keywords: Visual Studio Code extension, user-guide
+
+[id="adding-visual-studio-code-extension_{context}"]
+= Adding a Visual Studio Code extension to a workspace
+
+Devfiles in the devfile v2 format don't contain any IDE-specific plug-ins or Visual Studio Code extensions. Instead, the plug-ins and extensions are described in a specific meta-folder.
+
+[id="plug-in-registries"]
+== {prod-short} plug-in registries overview
+
+{prod-short} has a default plug-in and extensions registry installed in every {prod-short} instance. The Che-Theia IDE searches and installs the plug-ins and extensions from that location.
+
+Check this {prod-short} https://github.com/eclipse-che/che-plugin-registry[registry project] for an overview of default plug-ins, extensions, and source code. An online instance that is refreshed after every commit to the main branch, is located https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/[here]. You can use a different plug-in or extension registry for Che-Theia if you don't work in air gap mode: only the default registry is available there.
```

Comment:
In some che-docs topics, `link:` precedes URLs like this, but not in this PR. Is that okay?

Reply:
```suggestion
Check this {prod-short} https://github.com/eclipse-che/che-plugin-registry[registry project] for an overview of [the?] default plug-ins, extensions, and source code. An online instance that refreshes after every commit to the main branch is located https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/[here]. You can use a different plug-in or extension registry for Che-Theia if you don't work in air gap mode: only the default registry is available there.
```

Reply:
I'm having trouble making sense of the final sentence. 
- I don't see an "air gap mode" mentioned in the che-docs. Would it help to say "air gapped environment" or specify the parameters involved in setting up that environment, such as `airGapContainerRegistryHostname` et al?

Reply:
Using the definite article, "the default plug-ins", makes more sense if the "registry project" topic describes specific ones. Otherwise, if it's an abstract discussion of "plug-ins, extensions, and source code," the current wording is fine.

Reply:
'air gap' is also named 'restricted environment' sometimes. But you can find definition there https://en.wikipedia.org/wiki/Air_gap_(networking)

And it's used in Che (and docs)

Reply:
> 'air gap' is also named 'restricted environment' sometimes. But you can find definition there https://en.wikipedia.org/wiki/Air_gap_(networking)
> 
> And it's used in Che (and docs)

Although we should strive to use the terminology consistently, I'm not highlighting "air gap" as a problem. I'm calling out the use of "mode" because a user might expect to find an actual software *mode* when there isn't one. I'm guessing you mean "an air-gapped or restricted environment" instead.

Reply:
I'll accept any suggested changes but here suggested changes have questions so I never saw this kind of suggested changes.

>
Does "only the default registry is available there" mean "only the default registry is available in an air-gapped environment?" This seems like the inverse of what I would expect: That the user probably requires use of a custom plug-in or extension registry because they are in an air-gapped environment. The default URLs of the plug-ins and extensions might not be reachable from an air-gapped system.

There is no need to use a custom plug-in or extension registry. Che deploy a registry on the air-gapped environment and in that case it includes all the binaries so there is no need to fetch something externally. 

> The default URLs of the plug-ins and extensions might not be reachable from an air-gapped system.

everything is reachable in that case (we build and deploy an offline / air-gapped environment registry)

Reply:
wording fixed